// @bgd-labs/aave-address-book/dist/AaveV3Scroll.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/AaveV3Scroll.ts
var AaveV3Scroll_exports = {};
__export(AaveV3Scroll_exports, {
  AAVE_PROTOCOL_DATA_PROVIDER: () => AAVE_PROTOCOL_DATA_PROVIDER,
  ACL_ADMIN: () => ACL_ADMIN,
  ACL_MANAGER: () => ACL_MANAGER,
  ASSETS: () => ASSETS,
  CAPS_PLUS_RISK_STEWARD: () => CAPS_PLUS_RISK_STEWARD,
  CHAIN_ID: () => CHAIN_ID,
  COLLECTOR: () => COLLECTOR,
  CONFIG_ENGINE: () => CONFIG_ENGINE,
  DEFAULT_A_TOKEN_IMPL_REV_1: () => DEFAULT_A_TOKEN_IMPL_REV_1,
  DEFAULT_INCENTIVES_CONTROLLER: () => DEFAULT_INCENTIVES_CONTROLLER,
  DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1: () => DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1,
  DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1: () => DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1,
  EMISSION_MANAGER: () => EMISSION_MANAGER,
  E_MODES: () => E_MODES,
  FREEZING_STEWARD: () => FREEZING_STEWARD,
  L2_ENCODER: () => L2_ENCODER,
  ORACLE: () => ORACLE,
  POOL: () => POOL,
  POOL_ADDRESSES_PROVIDER: () => POOL_ADDRESSES_PROVIDER,
  POOL_ADDRESSES_PROVIDER_REGISTRY: () => POOL_ADDRESSES_PROVIDER_REGISTRY,
  POOL_CONFIGURATOR: () => POOL_CONFIGURATOR,
  PRICE_ORACLE_SENTINEL: () => PRICE_ORACLE_SENTINEL,
  RATES_FACTORY: () => RATES_FACTORY,
  UI_INCENTIVE_DATA_PROVIDER: () => UI_INCENTIVE_DATA_PROVIDER,
  UI_POOL_DATA_PROVIDER: () => UI_POOL_DATA_PROVIDER,
  WALLET_BALANCE_PROVIDER: () => WALLET_BALANCE_PROVIDER,
  WETH_GATEWAY: () => WETH_GATEWAY
});
module.exports = __toCommonJS(AaveV3Scroll_exports);
var POOL_ADDRESSES_PROVIDER = "0xdc5D225Df17df184d11015B91C4A10cd7834e2aC";
var POOL = "0xe37405957E371D07A1086a02ae33Ae443d4574F9";
var POOL_CONFIGURATOR = "0x81b161773CE43781A3E67d71f652E0E22f21e170";
var ORACLE = "0xD936362a1960ECE6c99a89F0a6f1104f0873826A";
var PRICE_ORACLE_SENTINEL = "0x0000000000000000000000000000000000000000";
var AAVE_PROTOCOL_DATA_PROVIDER = "0xa411Accec7000c52feE9bFeDaDc53E1CEF72d6d4";
var ACL_MANAGER = "0xAF4646B0131af8fc0DC435AF7F7d303Ac131E072";
var ACL_ADMIN = "0xc1ABF87FfAdf4908f4eC8dc54A25DCFEabAE4A24";
var COLLECTOR = "0x83a9EFE0485e7C227F29fe376132EfF746B45E96";
var DEFAULT_INCENTIVES_CONTROLLER = "0xB972ABEdB6047e0EfcfE1261c10F1a93C4daE29F";
var DEFAULT_A_TOKEN_IMPL_REV_1 = "0xA9E3fFb25C369e44862DD3e87Be4420abb879965";
var DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = "0x95eeA7A0b16C8ee3A923D3F5ebe6d77C0332084c";
var DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = "0x07c319CaCAcF9422fA18f3e7389A0BE1B64cAB86";
var EMISSION_MANAGER = "0x7417cB07350Cc50ad5d461648cA55E2405190887";
var L2_ENCODER = "0x0ffE481FBF0AE2282A5E1f701fab266aF487A97D";
var CAPS_PLUS_RISK_STEWARD = "0x36a92f2373400050656256B03Fa189145b708921";
var FREEZING_STEWARD = "0xCC3958a77c001a5608768537F628cA3b0A2A9386";
var CONFIG_ENGINE = "0x22913D4E21D44EF7662B118A6540450e25fE09a9";
var POOL_ADDRESSES_PROVIDER_REGISTRY = "0xb0633e01310a09C1Ee71a96c057DcF9c13fd6F62";
var RATES_FACTORY = "0xeF4d1F3F0dCe46C1Ad684F23Ea3aA542592A8cFe";
var UI_INCENTIVE_DATA_PROVIDER = "0x9E7DF170E44093d6738057157CA048794B02555d";
var UI_POOL_DATA_PROVIDER = "0x29CF7aC4Fc122085c0D4DE8894f878F0b141F799";
var WALLET_BALANCE_PROVIDER = "0x07D04EfAAA0Ac69D19d107795aF247C42Eb50F1C";
var WETH_GATEWAY = "0x93c597e5229f76679d7646cC9161B59Cc0DE78B9";
var CHAIN_ID = 534352;
var ASSETS = {};
var E_MODES = {};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAVE_PROTOCOL_DATA_PROVIDER,
  ACL_ADMIN,
  ACL_MANAGER,
  ASSETS,
  CAPS_PLUS_RISK_STEWARD,
  CHAIN_ID,
  COLLECTOR,
  CONFIG_ENGINE,
  DEFAULT_A_TOKEN_IMPL_REV_1,
  DEFAULT_INCENTIVES_CONTROLLER,
  DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1,
  DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1,
  EMISSION_MANAGER,
  E_MODES,
  FREEZING_STEWARD,
  L2_ENCODER,
  ORACLE,
  POOL,
  POOL_ADDRESSES_PROVIDER,
  POOL_ADDRESSES_PROVIDER_REGISTRY,
  POOL_CONFIGURATOR,
  PRICE_ORACLE_SENTINEL,
  RATES_FACTORY,
  UI_INCENTIVE_DATA_PROVIDER,
  UI_POOL_DATA_PROVIDER,
  WALLET_BALANCE_PROVIDER,
  WETH_GATEWAY
});
