// @bgd-labs/aave-address-book/dist/GovernanceV3Ethereum.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Ethereum.ts
var GovernanceV3Ethereum_exports = {};
__export(GovernanceV3Ethereum_exports, {
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  DATA_WAREHOUSE: () => DATA_WAREHOUSE,
  EMERGENCY_REGISTRY: () => EMERGENCY_REGISTRY,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  EXECUTOR_LVL_2: () => EXECUTOR_LVL_2,
  GOVERNANCE: () => GOVERNANCE,
  GOVERNANCE_POWER_STRATEGY: () => GOVERNANCE_POWER_STRATEGY,
  GOV_DATA_HELPER: () => GOV_DATA_HELPER,
  META_DELEGATE_HELPER: () => META_DELEGATE_HELPER,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER,
  VM_DATA_HELPER: () => VM_DATA_HELPER,
  VOTING_MACHINE: () => VOTING_MACHINE,
  VOTING_PORTAL_ETH_AVAX: () => VOTING_PORTAL_ETH_AVAX,
  VOTING_PORTAL_ETH_ETH: () => VOTING_PORTAL_ETH_ETH,
  VOTING_PORTAL_ETH_POL: () => VOTING_PORTAL_ETH_POL,
  VOTING_STRATEGY: () => VOTING_STRATEGY
});
module.exports = __toCommonJS(GovernanceV3Ethereum_exports);
var CROSS_CHAIN_CONTROLLER = "0xEd42a7D8559a463722Ca4beD50E0Cc05a386b0e1";
var GOVERNANCE = "0x9AEE0B04504CeF83A65AC3f0e838D0593BCb2BC7";
var PAYLOADS_CONTROLLER = "0xdAbad81aF85554E9ae636395611C58F7eC1aAEc5";
var VOTING_MACHINE = "0x617332a777780F546261247F621051d0b98975Eb";
var VOTING_PORTAL_ETH_ETH = "0xf23f7De3AC42F22eBDA17e64DC4f51FB66b8E21f";
var VOTING_PORTAL_ETH_AVAX = "0x33aCEf7365809218485873B7d0d67FeE411B5D79";
var VOTING_PORTAL_ETH_POL = "0x9b24C168d6A76b5459B1d47071a54962a4df36c3";
var PC_DATA_HELPER = "0xE3B770Dc4ae3f8bECaB3Ed12dE692c741603e16A";
var GOV_DATA_HELPER = "0x971c82c8316aD611904F95616c21ce90837f1856";
var VM_DATA_HELPER = "0x77976B51569896523EE215962Ee91ff236Fa50E8";
var META_DELEGATE_HELPER = "0x94363B11b37BC3ffe43AB09cff5A010352FE85dC";
var EMERGENCY_REGISTRY = "0x73C6Fb358dDA8e84D50e98A98F7c0dF32e15C7e9";
var GOVERNANCE_POWER_STRATEGY = "0xa198Fac58E02A5C5F8F7e877895d50cFa9ad1E04";
var EXECUTOR_LVL_1 = "0x5300A1a15135EA4dc7aD5a167152C01EFc9b192A";
var EXECUTOR_LVL_2 = "0x17Dd33Ed0e3dD2a80E37489B8A63063161BE6957";
var VOTING_STRATEGY = "0x5642A5A5Ec284B4145563aBF319620204aCCA7f4";
var DATA_WAREHOUSE = "0x1699FE9CaDC8a0b6c93E06B62Ab4592a0fFEcF61";
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CROSS_CHAIN_CONTROLLER,
  DATA_WAREHOUSE,
  EMERGENCY_REGISTRY,
  EXECUTOR_LVL_1,
  EXECUTOR_LVL_2,
  GOVERNANCE,
  GOVERNANCE_POWER_STRATEGY,
  GOV_DATA_HELPER,
  META_DELEGATE_HELPER,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER,
  VM_DATA_HELPER,
  VOTING_MACHINE,
  VOTING_PORTAL_ETH_AVAX,
  VOTING_PORTAL_ETH_ETH,
  VOTING_PORTAL_ETH_POL,
  VOTING_STRATEGY
});
