// @bgd-labs/aave-address-book/dist/GovernanceV3Polygon.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Polygon.ts
var GovernanceV3Polygon_exports = {};
__export(GovernanceV3Polygon_exports, {
  CL_EMERGENCY_ORACLE: () => CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  DATA_WAREHOUSE: () => DATA_WAREHOUSE,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  EXECUTOR_LVL_2: () => EXECUTOR_LVL_2,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER,
  VM_DATA_HELPER: () => VM_DATA_HELPER,
  VOTING_MACHINE: () => VOTING_MACHINE,
  VOTING_STRATEGY: () => VOTING_STRATEGY
});
module.exports = __toCommonJS(GovernanceV3Polygon_exports);
var CROSS_CHAIN_CONTROLLER = "0xF6B99959F0b5e79E1CC7062E12aF632CEb18eF0d";
var CL_EMERGENCY_ORACLE = "0xDAFA1989A504c48Ee20a582f2891eeB25E2fA23F";
var VOTING_MACHINE = "0xc8a2ADC4261c6b669CdFf69E717E77C9cFeB420d";
var PAYLOADS_CONTROLLER = "0x401B5D0294E23637c18fcc38b1Bca814CDa2637C";
var PC_DATA_HELPER = "0xE3B770Dc4ae3f8bECaB3Ed12dE692c741603e16A";
var VM_DATA_HELPER = "0x77976B51569896523EE215962Ee91ff236Fa50E8";
var EXECUTOR_LVL_1 = "0xDf7d0e6454DB638881302729F5ba99936EaAB233";
var EXECUTOR_LVL_2 = "0x0000000000000000000000000000000000000000";
var VOTING_STRATEGY = "0x59e6CAD5d7E7b9A26a45a1d1E74C7aF008170042";
var DATA_WAREHOUSE = "0xf41193E25408F652AF878c47E4401A01B5E4B682";
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER,
  DATA_WAREHOUSE,
  EXECUTOR_LVL_1,
  EXECUTOR_LVL_2,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER,
  VM_DATA_HELPER,
  VOTING_MACHINE,
  VOTING_STRATEGY
});
