// @bgd-labs/aave-address-book/dist/MiscEthereum.mjs
// src/ts/MiscEthereum.ts
var AAVE_ECOSYSTEM_RESERVE_CONTROLLER = "0x3d569673dAa0575c936c7c67c4E6AedA69CC630C";
var ECOSYSTEM_RESERVE = "0x25F2226B597E8F9514B3F68F00f494cF4f286491";
var PROXY_ADMIN_LONG = "0x86C3FfeE349A7cFf7cA88C449717B1b133bfb517";
var AAVE_SWAPPER = "0x3ea64b1C0194524b48F9118462C8E9cd61a243c7";
var AAVE_POL_ETH_BRIDGE = "0x1C2BA5b8ab8e795fF44387ba6d251fa65AD20b36";
var PARASWAP_FEE_CLAIMER = "0x9abf798f5314BFd793A9E57A654BEd35af4A1D60";
var TRANSPARENT_PROXY_FACTORY = "0xB4e496f70602fE2AC6Ae511D028BA4D194773B29";
var PROXY_ADMIN = "0xD3cF979e676265e4f6379749DECe4708B9A22476";
var CREATE_3_FACTORY = "0xcc3C54B95f3f1867A43009B80ed4DD930E3cE2F7";
var PROTOCOL_GUARDIAN = "0xCA76Ebd8617a03126B6FB84F9b1c1A0fB71C2633";
var AAVE_MERKLE_DISTRIBUTOR = "0xa88c6D90eAe942291325f9ae3c66f3563B93FE10";
export {
  AAVE_ECOSYSTEM_RESERVE_CONTROLLER,
  AAVE_MERKLE_DISTRIBUTOR,
  AAVE_POL_ETH_BRIDGE,
  AAVE_SWAPPER,
  CREATE_3_FACTORY,
  ECOSYSTEM_RESERVE,
  PARASWAP_FEE_CLAIMER,
  PROTOCOL_GUARDIAN,
  PROXY_ADMIN,
  PROXY_ADMIN_LONG,
  TRANSPARENT_PROXY_FACTORY
};
