// @bgd-labs/aave-address-book/dist/GovernanceV3Ethereum.mjs
// src/ts/GovernanceV3Ethereum.ts
var CROSS_CHAIN_CONTROLLER = "0xEd42a7D8559a463722Ca4beD50E0Cc05a386b0e1";
var GOVERNANCE = "0x9AEE0B04504CeF83A65AC3f0e838D0593BCb2BC7";
var PAYLOADS_CONTROLLER = "0xdAbad81aF85554E9ae636395611C58F7eC1aAEc5";
var VOTING_MACHINE = "0x617332a777780F546261247F621051d0b98975Eb";
var VOTING_PORTAL_ETH_ETH = "0xf23f7De3AC42F22eBDA17e64DC4f51FB66b8E21f";
var VOTING_PORTAL_ETH_AVAX = "0x33aCEf7365809218485873B7d0d67FeE411B5D79";
var VOTING_PORTAL_ETH_POL = "0x9b24C168d6A76b5459B1d47071a54962a4df36c3";
var PC_DATA_HELPER = "0xE3B770Dc4ae3f8bECaB3Ed12dE692c741603e16A";
var GOV_DATA_HELPER = "0x971c82c8316aD611904F95616c21ce90837f1856";
var VM_DATA_HELPER = "0x77976B51569896523EE215962Ee91ff236Fa50E8";
var META_DELEGATE_HELPER = "0x94363B11b37BC3ffe43AB09cff5A010352FE85dC";
var EMERGENCY_REGISTRY = "0x73C6Fb358dDA8e84D50e98A98F7c0dF32e15C7e9";
var GOVERNANCE_POWER_STRATEGY = "0xa198Fac58E02A5C5F8F7e877895d50cFa9ad1E04";
var EXECUTOR_LVL_1 = "0x5300A1a15135EA4dc7aD5a167152C01EFc9b192A";
var EXECUTOR_LVL_2 = "0x17Dd33Ed0e3dD2a80E37489B8A63063161BE6957";
var VOTING_STRATEGY = "0x5642A5A5Ec284B4145563aBF319620204aCCA7f4";
var DATA_WAREHOUSE = "0x1699FE9CaDC8a0b6c93E06B62Ab4592a0fFEcF61";
export {
  CROSS_CHAIN_CONTROLLER,
  DATA_WAREHOUSE,
  EMERGENCY_REGISTRY,
  EXECUTOR_LVL_1,
  EXECUTOR_LVL_2,
  GOVERNANCE,
  GOVERNANCE_POWER_STRATEGY,
  GOV_DATA_HELPER,
  META_DELEGATE_HELPER,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER,
  VM_DATA_HELPER,
  VOTING_MACHINE,
  VOTING_PORTAL_ETH_AVAX,
  VOTING_PORTAL_ETH_ETH,
  VOTING_PORTAL_ETH_POL,
  VOTING_STRATEGY
};
