// @bgd-labs/aave-address-book/dist/GovernanceV3Gnosis.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Gnosis.ts
var GovernanceV3Gnosis_exports = {};
__export(GovernanceV3Gnosis_exports, {
  CL_EMERGENCY_ORACLE: () => CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  EXECUTOR_LVL_2: () => EXECUTOR_LVL_2,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER
});
module.exports = __toCommonJS(GovernanceV3Gnosis_exports);
var CROSS_CHAIN_CONTROLLER = "0x8Dc5310fc9D3D7D1Bb3D1F686899c8F082316c9F";
var CL_EMERGENCY_ORACLE = "0xF937ffAeA1363e4Fa260760bDFA2aA8Fc911F84D";
var PAYLOADS_CONTROLLER = "0x9A1F491B86D09fC1484b5fab10041B189B60756b";
var PC_DATA_HELPER = "0xF1c11BE0b4466728DDb7991A0Ac5265646ec9672";
var EXECUTOR_LVL_1 = "0x1dF462e2712496373A347f8ad10802a5E95f053D";
var EXECUTOR_LVL_2 = "0x0000000000000000000000000000000000000000";
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1,
  EXECUTOR_LVL_2,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER
});
